### PR TITLE
Use correct dummy world for BlockRenderer6343

### PR DESCRIPTION
**Pull request by Lyfts**
Closes https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/16755

No idea how it gets to this crash, but the packet that's causing it is hidden behind a [!GregTechAPI.isDummyWorld()](https://github.com/GTNewHorizons/GT5-Unofficial/blob/b0364a9c8630d217daf1d9b8379253d4eeeb0b76/src/main/java/gregtech/api/metatileentity/implementations/MTEEnhancedMultiBlockBase.java#L64) check and this is currently checking for the wrong one.

### CODE REVIEW

File: src/main/java/gregtech/api/GregTechAPI.java [hunk 1]
```diff
@@ -289,7 +291,10 @@ public class GregTechAPI {
...
 
         dummyWorlds.add(GTDummyWorld.class);
-        tryAddDummyWorld("blockrenderer6343.client.world.DummyWorld");
+
+        if (Mods.BlockRenderer6343.isModLoaded()) {
+            dummyWorlds.add(TrackedDummyWorld.class);
+        }
     }
 
     private static void tryAddDummyWorld(String className) {
```
Comment (Alexdoru) [inline src/main/java/gregtech/api/GregTechAPI.java:300]: delete this method below it's now unused

**Review by Alexdoru — CHANGES_REQUESTED** [review]

**Review by Alexdoru — APPROVED** [review]

**Comment by Lyfts** [comment]
Idk why tf this causes every single test to fail... It builds (without tests) and runs fine locally, but it would make a lot more sense to add the dummy world from BlockRenderer anyways.

**Comment by Alexdoru** [comment]
> Idk why tf this causes every single test to fail... It builds (without tests) and runs fine locally, but it would make a lot more sense to add the dummy world from BlockRenderer anyways.

you need to apply spotless

**Comment by Lyfts** [comment]
It's not spotless

**Comment by Lyfts** [comment]
It causes all the unit tests to fail with an `ExceptionInInitializerError` but builds fine otherwise for some reason?  Anyways closed in favor of https://github.com/GTNewHorizons/BlockRenderer6343/pull/21

**Pull request closed without merging**